onadata/libs/serializers/data_serializer.py

```diff
@@ -46,7 +46,7 @@ def create_submission(request, username, data_dict, xform_id):
     """
     Returns validated data object instances
     """
-    xml_string = dict2xform(data_dict, xform_id)
+    xml_string = dict2xform(data_dict, xform_id, root='data')
```

Comment:
What happens to older forms that still use the filename of the XLSForm that is in our historical data?

Reply:
That's something I did not initially consider... I'll look into retrieving the root node name from the form. I believe pyxform provides this information on the survey object, I'll confirm and update the PR

Reply:
Updated the PR. We now try to retrieve the root node name from the `XForm` if a username is passed alongside the `form_id`. This should ensure all new submissions have the correct root node.